ceremonies/src/lib.rs

```diff
@@ -1839,11 +1845,15 @@ impl<T: Config> OnCeremonyPhaseChange for Pallet<T> {
 				let cindex = <encointer_scheduler::Pallet<T>>::current_ceremony_index();
 				// Clean up with a time delay, such that participants can claim their UBI in the following cycle.
 				if cindex > Self::reputation_lifetime() {
-					Self::purge_registry(cindex - Self::reputation_lifetime() - 1);
+					Self::purge_registry(
+						cindex.saturating_sub(Self::reputation_lifetime()).saturating_sub(1),
+					);
 				}
-				let inactives = Self::get_inactive_communities(
-					<encointer_scheduler::Pallet<T>>::current_ceremony_index() - 1,
-					Self::inactivity_timeout(),
+				let inactives = Self::update_inactivity_counters(
+					// check inactivity with delay, such that rewards can be claimed before the inactivity check happens
+					<encointer_scheduler::Pallet<T>>::current_ceremony_index().saturating_sub(2),
+					// compensate for the delay
+					Self::inactivity_timeout().saturating_add(1),
```

Comment:
Ah, nice, I see the fix!

However, this kind of seems like we are doing a hack here, maybe we could consider two options?

* we could also just move this to assignment phase
* In the tests I see that we purge at `inactivity_timout + 1`, so from my point of view, we can change the definition here and remove the `saturating_add` to compensate, this would also make it more easy to read.

@brenzi your input?

Reply:
I see what you mean. I was thinking about moving it to the assignment phase but I dont think it makes sense from a UX perspective. Then you could register and suddenly the community is gone.
We could just remove the add. I just wanted to be safe that there are no issues with newly created communities in a setting where the inactivity timeout is 1.
What exactly do you mean with `we can change the definition here and remove the saturating_add to compensate`?

Reply:
However, the inactivity timeout is global, set by encointer governance, hence I suppose we could just disallow 1 in the setter.

Still this is another discussion, we can leave it like that for now, IMO.

Reply:
Isn't the UX even worse now? If we check cindex-2 this means that there may have been 5 idle ceremonies plus one active full ceremony cycle with legit attendees - but still we purge, because of idle 5 ceremonies before the last one.

I think I'd prefer moving this check to the start of assigning phase. No harm if people registered in vain. Big harm if they attended in vain

Reply:
yeah this is true.